src/Verifier/SAW/SharedTerm.hs

```diff
@@ -895,6 +895,38 @@ scImport sc t0 =
 --------------------------------------------------------------------------------
 -- Instantiating variables
 
+-- | The second argument is a function that takes the number of
+-- enclosing lambdas and the de Bruijn index of the variable,
+-- returning the new term to replace it with.
+instantiateLocalVars ::
+  SharedContext ->
+  (DeBruijnIndex -> DeBruijnIndex -> IO Term) ->
+  DeBruijnIndex -> Term -> IO Term
+instantiateLocalVars sc f initialLevel t0 =
+  do cache <- newCache
+     let ?cache = cache in go initialLevel t0
+  where
+    go :: (?cache :: Cache IO (TermIndex, DeBruijnIndex) Term) =>
+          DeBruijnIndex -> Term -> IO Term
+    go l t =
+      case t of
+        Unshared tf -> go' l tf
+        STApp{ stAppIndex = tidx, stAppFreeVars = _fv, stAppTermF = tf}
+          --   | fv == emptyBitSet -> return t -- closed terms map to themselves
```

Comment:
I think you commented out the actual special case...

Reply:
Oops! I was doing some performance tests both ways and apparently forgot to change it back.